org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java

```diff
@@ -86,6 +99,38 @@ public IMethodCoverage getCoverage() {
 		return coverage;
 	}
 
+	/**
+	 * {@link MethodNode#accept(MethodVisitor)}
+	 */
+	@Override
+	public void accept(final MethodNode methodNode,
+			final MethodVisitor methodVisitor) {
+		this.ignored.clear();
+		for (final IFilter filter : FILTERS) {
+			filter.filter(methodNode, this);
+		}
+
+		for (final TryCatchBlockNode n : methodNode.tryCatchBlocks) {
+			n.accept(methodVisitor);
+		}
+		for (int i = 0; i < methodNode.instructions.size(); i++) {
```

Comment:
In fact I was wrong - default implementation in `org.objectweb.asm.tree.MethodNode#accept(org.objectweb.asm.MethodVisitor)` does not construct array, don't know from where opposite came to my mind. Still wondering how to prevent such regressions, especially given that `MethodNode#instructions` is accessible by filters now.